### PR TITLE
Update everything for Xcode 8.1

### PR DESCRIPTION
**Pull request by rastersize**
Fixes a bunch of warnings and updates to Swift 3.0.1 for the demo project.

@spotify/objc-dev

### CODE REVIEW

File: spotify_os.xcconfig [hunk 1]
```diff
@@ -62,6 +62,15 @@ HF_IGNORED_WARNINGS_0820 = $(HF_IGNORED_WARNINGS_0810)
 
 WARNING_CFLAGS = -Weverything -Wno-error=deprecated -Wno-objc-missing-property-synthesis -Wno-gnu-conditional-omitted-operand -Wno-gnu -Wno-reserved-id-macro -Wno-auto-import -Wno-missing-variable-declarations -Wno-vla -Werror $(HF_IGNORED_WARNINGS_$(XCODE_VERSION_MINOR))
 
+// Apple LLVM 8.0 - Warnings - All languages
```
Comment (8W9aG) [inline spotify_os.xcconfig]: Shouldn't these be done in the ci module?

Comment (JohnSundell) [inline spotify_os.xcconfig]: Don't think we're using it yet, right @rastersize?

Comment (rastersize) [inline spotify_os.xcconfig]: Yeah true. Will open a PR for ios-ci for those changes and then we can do bump.

Comment (rastersize) [inline spotify_os.xcconfig]: @JohnSundell You have a better memory than I do. We’re not actually using ios-ci for Hubs (more like ios-ci v2.0). So will have to update in both places.

@8W9aG I’ve sent a PR for ios-ci: https://github.com/spotify/ios-ci/pull/19

**Pull request merged**